src/System.Text.Json/src/System/Text/Json/JsonHelpers.Date.cs

```diff
@@ -57,6 +57,9 @@ public static bool TryParseAsISO(ReadOnlySpan<byte> source, out DateTimeOffset v
         // YYYY-MM-DDThh:mm:ss.sTZD (eg 1997-07-16T19:20:30.45-0100)
         // YYYY-MM-DDThh:mm:ss.sTZD (eg 1997-07-16T19:20:30.45+01)
         // YYYY-MM-DDThh:mm:ss.sTZD (eg 1997-07-16T19:20:30.45-01)
+        // ---------------------------------
+        // The - and / values may be used interchangeably as input here eg:
```

Comment:
The default behavior of parsing `DateTime{Offset}` is restricted to the ISO 8601 format which doesn't support slashes and is documented as such: https://github.com/dotnet/corefx/blob/fb81332b9643ecf8c0bb34d7938da97aa5e13e03/src/System.Text.Json/src/System/Text/Json/Reader/Utf8JsonReader.TryGet.cs#L285-L287

https://en.wikipedia.org/wiki/ISO_8601

I see why support for non-ISO date times could be useful, but it would make more sense to build such support on top (including all sorts of other custom formats - such as G, R (RFC 1123), l), and we shouldn't modify the default behavior since it makes it harder to clearly document the behavior.

The current workaround for callers is to get the raw bytes (or the JSON string) and parse it themselves (maybe by leveraging the existing `Utf8Parser` APIs or the `DateTime.Parse` methods):
https://github.com/dotnet/corefx/blob/fb81332b9643ecf8c0bb34d7938da97aa5e13e03/src/Common/src/CoreLib/System/Buffers/Text/Utf8Parser/Utf8Parser.Date.cs#L33
https://github.com/dotnet/corefx/blob/fb81332b9643ecf8c0bb34d7938da97aa5e13e03/src/Common/src/CoreLib/System/DateTime.cs#L1198

cc @layomia

Reply:
@thegreatco, if you have a motivating scenario for this change, could you please file an issue and we can continue the discussion over there (especially if the workarounds don't work for you)? We should think about solving this problem for vNext and design the custom capability for other formats (for example numbers in hex format, or ones with comma delimiters).